### ISSUE TITLE
Autowiring Actions fail

### ISSUE THREAD

**Issue by felicitus** [api-platform/core#1334]
Hi,

I'm upgrading my project to Symfony 3.3 and the new api-platform. However, when I attempt to autowire an action, I receive the error that no services are passed to my action. I have enabled service autowiring.

It is noteworthy that the action is instanciated by Symfonys `ControllerResolver` line 202, where it creates a new instance of my class without any parameters appended.

Do you have any idea where I could start looking?

Entity:
```
/**
 * @ORM\Entity
 * @ORM\Table(
 *      name="PartKeeprUser",
 *      uniqueConstraints={@ORM\UniqueConstraint(name="username_provider", columns={"username", "provider_id"})})
 * @TargetService(uri="/api/users")
 *
 * @ApiResource(
 *     collectionOperations={
 *          "user_preferences"={"route_name"="user_get_preferences"}
 *     },
 *     attributes={"filters"={"doctrine_reflection_service.search_filter"}}
 *  )
 */
class User extends BaseEntity implements UserInterface, EquatableInterface
{
```

routing.yml:
```
auth_actions:
    resource: "@PartKeeprAuthBundle/Action"
    type:     annotation
```

Action:
```
<?php

namespace PartKeepr\AuthBundle\Action;

use PartKeepr\AuthBundle\Services\UserPreferenceService;
use PartKeepr\AuthBundle\Services\UserService;
use PartKeepr\AuthBundle\Entity\User;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Serializer\Serializer;

/**
 * Returns the tree root node.
 */
class GetPreferencesAction
{
    /**
     * @var UserService
     */
    private $userService;

    /**
     * @var UserPreferenceService
     */
    private $userPreferenceService;

    /**
     * @var Serializer
     */
    private $serializer;

    public function __construct(
       /* UserService $userService,
        UserPreferenceService $userPreferenceService,*/
        Serializer $serializer
    ) {
        /*$this->userService = $userService;
        $this->userPreferenceService = $userPreferenceService;*/
        $this->serializer = $serializer;
    }

    /**
     * Retrieves a collection of resources.
     *
     * @Route(
     *     name="user_get_preferences",
     *     path="/api/user_preferences",
     *     defaults={"_api_resource_class"=User::class, "_api_collection_operation_name"="user_preferences"}
     * )
     * @Method("GET")
     * @param Request $request
     * @return JsonResponse
     */

    public function __invoke($data, Request $request)
    {
        die("FOO");
        $user = $this->userService->getUser();

        $preferences = $this->userPreferenceService->getPreferences($user);

        /*
         * @var ResourceInterface $resourceType
         */
        $serializedData = $this->serializer->normalize(
            $preferences,
            'json'
        );

        return new JsonResponse($serializedData);
    }
}
```

**Comment by OnekO** [comment]
Same problem here with a very similar configuration. Any advice?

**Comment by OnekO** [comment]
Solved thanks to this: [https://github.com/api-platform/api-platform/issues/317#issuecomment-316436505](https://github.com/api-platform/api-platform/issues/317#issuecomment-316436505)

Just add something like
`AppBundle\Action\:
        resource: '../../src/AppBundle/Action'
        public: true
        autowire: true
        tags: ['controller.service_arguments']`
to your services.yml

**Comment by dunglas** [comment]
Fixed in API Platform 2.1.
